src/main/java/hudson/plugins/gradle/injection/MavenAbstractBuildRunListener.java

```diff
@@ -0,0 +1,18 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.Extension;
+import hudson.maven.AbstractMavenBuild;
+import hudson.model.TaskListener;
+import hudson.model.listeners.RunListener;
+
+@Extension
+public class MavenAbstractBuildRunListener extends RunListener<AbstractMavenBuild<?, ?>> implements MavenInjectionAware {
+
+    @Override
+    public void onStarted(AbstractMavenBuild abstractMavenBuild, TaskListener listener) {
+        super.onInitialize(abstractMavenBuild);
+
+        setPreparedMavenProperties(abstractMavenBuild, listener);
```

Comment:
Why this method is located in `MavenInjectionAware`? This logic is very specific to the `RunListener`. I don't think that we want it to leak into the shared code. If you want to share it between different listeners, then please create a custom abstract listener that can be used to share that code.

Reply:
I also thought about creating a separate abstract class for this, but then thought it might be too much and since we already have a Maven injection specific interface, it made sense to keep it there (it also uses variables from it, so everything maven injection related is in one place).

But perhaps you're right and extracting it would be better. I'll update this.

Reply:
Hah, we already extend EnvironmentContributor, so cannot extend another one here :)

So either that interface or a util class

Reply:
This is now extracted into one listener implementation